RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEGenericESProducer.cc

```diff
@@ -55,7 +57,7 @@ PixelCPEGenericESProducer::PixelCPEGenericESProducer(const edm::ParameterSet& p)
   UseErrorsFromTemplates_ = p.getParameter<bool>("UseErrorsFromTemplates");
 
   pset_ = p;
-  auto c = setWhatProduced(this, myname);
+  auto c = setWhatProduced(this, CPEgenericMode_);
```

Comment:
Do you need all users of the `PixelClusterParameterEstimator` from this ESProducer to use either the `PixelCPEGenericForBricked` or `PixelCPEGeneric`? Or does it make sense for some to use one and some the other?

If the answer is the former (i.e. the two are mutually exclusive), not using the `ComponentName` for specifying which one to produce would relieve from customizing the users because the product label would stay the same.

Reply:
This point has been discussed with @makortel and @mmusich.
There are pros and cons with both the approaches (e.g. keeping the code as it is or following the suggestion by Matti)
We keep the current implementation to be reviewed by the reco and offline managers